### ISSUE TITLE
de_DE bank provider does not honor BIC specifications by German authority

### ISSUE THREAD

**Issue by kkotenko** [joke2k/faker#1823]
* Faker version: 16.6.0
* OS: Linux

In Germany, the norms for EBICS/SWIFT processing are de facto defined by the [Deutsche Kreditwirtschaft](https://en.wikipedia.org/wiki/German_Banking_Industry_Committee). 

In its specification of the SWIFT number, the DK takes a slightly more restrictive approach to a valid SWIFT number: it specifies it with the regular expression `[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}`. (In other words, there must not be a 0 or 1 in the 7th place and no O in the 8th place.) Source for this specification is [Appendix 3 Data Formats 3.6.pdf](https://www.ebics.de/de/datenformate), section 2.3.1

Faker does not currently honor this restriction. This clashes with the verification against an xsd schema file provided by the DK, which third-party libraries perform. Thus, Faker is not able to provide data which is consistently accepted by those libraries.

### Steps to reproduce

```python
from faker import Faker
import re

Faker.seed(0)

fake = Faker(locale='de_DE')

if __name__ == '__main__':

    for i in range(100):
        swift = fake.swift()
        if not re.match(r"[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}", swift):
            print(f"{i}:{swift}")
```

### Expected behavior

The output should be empty.

### Actual behavior

The output is 
```
14:NBDZDEJOC8E
26:WJMXDE0F
28:UOMWDE0C
31:KJOBDE0M9FI
43:JMKJDE0G
55:ZCQBDEEO
59:RTNPDEYOBAL
69:AQVSDENOF67
70:QJDEDE11FG0
```

**Comment by fcurella** [comment]
Thank you for the report @kkotenko! Could you submit a Pull Request?

**Comment by fcurella** [comment]
merged in [d93d8a0](https://github.com/joke2k/faker/commit/d93d8a0d42b7a92d8b386d73c85886b2633124e7)
